### ISSUE TITLE
Mapnik.VectorTile.painted() doesn't do what we want it to do

### ISSUE THREAD

**Issue by mapsam** [mapbox/tilelive-bridge#85]
**TLDR**

`mapnik.VectorTile.painted()` doesn't do exactly what we want any longer, which means our check [here](https://github.com/mapbox/tilelive-bridge/blob/2c4de98597fc61f9fc5315c656d6964b78459c86/index.js#L286) isn't as strong as expected.

**Problem**

When generating tiles and copying them (via tilelive), we have a check in the pyramid tiling scheme to ensure we should continue rendering and copying child tiles. This check is [here](https://github.com/mapbox/tilelive/blob/0cb09a3313167e38f00a4dc04e783ca132a50a16/lib/stream-pyramid.js#L149). 

There are some situations where we could have a datasource (geojson for this example) that has data in the z0 tile, but has no data in the z1 tile. This means tilelive-bridge checks `vtile.painted()` and sets the `x-tilelive-contains-data` header to `false`. If this is set to `false`, the rest of the tiling scheme is _skipped_, resulting in no tiles.

Node Mapnik essentially [uses `empty()` to check for painted tiles](https://github.com/mapbox/mapnik-vector-tile/blob/master/src/vector_tile_tile.hpp#L116-L119) via mapnik-vector-tile, which is exactly what we don't want. Right now this z1 tile returns `false` for `empty()`, which tells tilelive to stop the copy process and skip the rest of the tiles. We need to know if the source data is there (even if tile features are not) so we can continue to tile the children.

**Why**

Currently, I'm unable to figure out _why_ a datasource in Node Mapnik would add data at z0 but not at z1. 

**Solution**

A couple solutions come to mind:

* [time intensive] improve the `painted()`, `solid()`, and `empty()` architecture in Mapnik. cc @flippmoke 
* [quick fix, potentially at CPU expense] don't check for painted/empty tiles since we already have a relatively solid pyramid built via mapnik-omnivore and spherical mercator. This either means we always set `x-tilelive-contains-data` to `true` or remove the `if` statement in tilelive.

cc @rclark @GretaCB @springmeyer @yhahn @flippmoke @captainbarbosa

**Comment by yhahn** [comment]
@mapsam thanks for digging into this -- it sounds like something we should look into further and try to fix.

Removing this check in the tiling pyramid scheme is definitely not an option -- it's used quite a bit, especially for very large/diverse datasets.

**Comment by mapsam** [comment]
This issue continues to come up. An immediate medium-term solution is to change our usage of the pyramid tiling scheme in mapbox-tile-copy to "scanline", which will ensure each tile has no geometry before being tossed, no matter the parent zoom. Some quick benchmarks on the [NED 10m_lakes](http://www.naturalearthdata.com/downloads/10m-physical-vectors/10m-lakes/) file:

"scanline": `25` seconds average copy time
"pyramid": `24.66` seconds average copy time
